### ISSUE TITLE
Curriculum challenges with external links - what to do with them?

### ISSUE THREAD

**Issue by RandellDawson** [freeCodeCamp/freeCodeCamp#40749]
In a recent internal discussion, the team wanted to understand how many curriculum challenges currently have external links (outside of freeCodeCamp) where we may be relying on the external link's information to teach something needed to solve the challenge.  The discussion came about after a translator asked if an external ink could be changed to an external site with relevant information written in the corresponding language to be translated.

The following is a list of such challenges and their corresponding links.  As almost every Rosetta Code challenge has a link to the original Rosetta Code site (to give source credit), the only Rosetta Code challenges listed below contain external links that are not Rosetta Code site pages but are present in the descriptions.

The purpose of this issue is to discuss/decide what to do for each challenge link listed.  The choices are:
- Leave existing link in challenge
- Rewrite challenge so external link is not needed
- Link to an existing freeCodeCamp news article containing the relevant information contained in the externally linked page.
- Create new freeCodeCamp news articles containing the relevant information contained in the externally linked page.

<details><summary> List of files(collapsed list as it's a long list) </summary>

- [x] #46414
`http://mongoosejs.com/docs/guide.html`
 [curriculum/challenges/english/05-apis-and-microservices/mongodb-and-mongoose/perform-classic-updates-by-running-find-edit-then-save.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/05-apis-and-microservices/mongodb-and-mongoose/perform-classic-updates-by-running-find-edit-then-save.md)
`Instructions Link: https://mongoosejs.com/docs/schematypes.html#Mixed`
- [x] [curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/use-a-template-engines-powers.md)
`https://github.com/pugjs/pug`
- [x] [curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/authentication-strategies.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/authentication-strategies.md)
`http://passportjs.org/`
- [x] [curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implement-the-serialization-of-a-passport-user.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implement-the-serialization-of-a-passport-user.md)
`https://www.mongodb.com/cloud/atlas`
- [x] [curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication-ii.md](https://github.com/jaredhanson/passport-github/)
`https://github.com/jaredhanson/passport-github/`
- [ ] [curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implementation-of-social-authentication.md), we have decided to swap the link with freeCodeCamp article, the link should be kept until a related article is created 
`https://github.com/settings/developers`
- [x] [curriculum/challenges/english/01-responsive-web-design/basic-css/import-a-google-font.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/01-responsive-web-design/basic-css/import-a-google-font.md)
`https://fonts.google.com/`
- [x] ~~[curriculum/challenges/english/01-responsive-web-design/basic-css/use-hex-code-for-specific-colors.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/01-responsive-web-design/basic-css/use-hex-code-for-specific-colors.md)
`https://en.wikipedia.org/wiki/Hexadecimal`
`https://en.wikipedia.org/wiki/RGB_color_model`~~
- [x] ~~[curriculum/challenges/english/01-responsive-web-design/css-flexbox/align-elements-using-the-justify-content-property.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/01-responsive-web-design/css-flexbox/align-elements-using-the-justify-content-property.md)
`https://www.w3.org/TR/css-flexbox-1/images/flex-direction-terms.svg`~~
- [x] ~~[curriculum/challenges/english/01-responsive-web-design/applied-visual-design/learn-about-complementary-colors.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/01-responsive-web-design/applied-visual-design/learn-about-complementary-colors.md)
`https://en.wikipedia.org/wiki/Color_model`~~
- [x] ~~[curriculum/challenges/english/01-responsive-web-design/responsive-web-design-projects/build-a-tribute-page.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/01-responsive-web-design/responsive-web-design-projects/build-a-tribute-page.md)
`https://en.wikipedia.org/wiki/User_story`~~
- [x] ~~[curriculum/challenges/english/01-responsive-web-design/responsive-web-design-projects/build-a-technical-documentation-page.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/01-responsive-web-design/responsive-web-design-projects/build-a-technical-documentation-page.md)
`https://en.wikipedia.org/wiki/User_story`~~
- [x] ~~[curriculum/challenges/english/01-responsive-web-design/responsive-web-design-projects/build-a-product-landing-page.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/01-responsive-web-design/responsive-web-design-projects/build-a-product-landing-page.md)
`https://en.wikipedia.org/wiki/User_story`~~
- [x] ~~[curriculum/challenges/english/01-responsive-web-design/responsive-web-design-projects/build-a-personal-portfolio-webpage.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/01-responsive-web-design/responsive-web-design-projects/build-a-personal-portfolio-webpage.md)
`https://en.wikipedia.org/wiki/User_story`~~
- [x] ~~[curriculum/challenges/english/01-responsive-web-design/responsive-web-design-projects/build-a-survey-form.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/01-responsive-web-design/responsive-web-design-projects/build-a-survey-form.md)
`https://en.wikipedia.org/wiki/User_story`~~
- [x] [curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-light-bright-app.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-light-bright-app.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-wikipedia-viewer.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-wikipedia-viewer.md)
`https://en.wikipedia.org/wiki/User_story`
`https://en.wikipedia.org/wiki/Special:Random`
`https://www.mediawiki.org/wiki/API:Main_page`
`https://en.wikipedia.org/wiki/Special:ApiSandbox#action=query&#x26;titles=Main%20Page&#x26;prop=revisions&#x26;rvprop=content&#x26;format=jsonfm`
- [x] [curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-recipe-box.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-recipe-box.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/10-coding-interview-prep/take-home-projects/use-the-twitch-json-api.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/take-home-projects/use-the-twitch-json-api.md)
`https://en.wikipedia.org/wiki/User_story`
`https://dev.twitch.tv/docs/api/reference/#get-streams">https://dev.twitch.tv/docs/api/reference/#get-streams`
`https://api.twitch.tv/helix`
- [x] [curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-simon-game.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-simon-game.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/10-coding-interview-prep/take-home-projects/p2p-video-chat-application.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/take-home-projects/p2p-video-chat-application.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/10-coding-interview-prep/take-home-projects/map-data-across-the-globe.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/take-home-projects/map-data-across-the-globe.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/10-coding-interview-prep/take-home-projects/manage-a-book-trading-club.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/take-home-projects/manage-a-book-trading-club.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-the-game-of-life.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-the-game-of-life.md)
`https://en.wikipedia.org/wiki/User_story`
`https://www.youtube.com/watch?v=E8kUJL04ELA`
`https://en.wikipedia.org/wiki/Conway%27s_Game_of_Life`
- [x] [curriculum/challenges/english/10-coding-interview-prep/take-home-projects/show-the-local-weather.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/take-home-projects/show-the-local-weather.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-nightlife-coordination-app.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-nightlife-coordination-app.md)
`https://www.yelp.com/developers/documentation/v3`
- [x] [curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-tic-tac-toe-game.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-tic-tac-toe-game.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-pong-game.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-pong-game.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-roguelike-dungeon-crawler-game.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-roguelike-dungeon-crawler-game.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/10-coding-interview-prep/take-home-projects/show-national-contiguity-with-a-force-directed-graph.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/take-home-projects/show-national-contiguity-with-a-force-directed-graph.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-pinterest-clone.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-pinterest-clone.md)
`https://en.wikipedia.org/wiki/User_story`
`https://masonry.desandro.com/">Masonry.js`
- [x] [curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-camper-leaderboard.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/take-home-projects/build-a-camper-leaderboard.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] ~~[curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-89-roman-numerals.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-89-roman-numerals.md)
`https://projecteuler.net/about=roman_numerals`~~
- [x] [curriculum/challenges/english/04-data-visualization/data-visualization-projects/visualize-data-with-a-bar-chart.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/04-data-visualization/data-visualization-projects/visualize-data-with-a-bar-chart.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/04-data-visualization/data-visualization-projects/visualize-data-with-a-treemap-diagram.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/04-data-visualization/data-visualization-projects/visualize-data-with-a-treemap-diagram.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/04-data-visualization/data-visualization-projects/visualize-data-with-a-heat-map.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/04-data-visualization/data-visualization-projects/visualize-data-with-a-heat-map.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/04-data-visualization/data-visualization-projects/visualize-data-with-a-scatterplot-graph.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/04-data-visualization/data-visualization-projects/visualize-data-with-a-scatterplot-graph.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/04-data-visualization/data-visualization-projects/visualize-data-with-a-choropleth-map.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/04-data-visualization/data-visualization-projects/visualize-data-with-a-choropleth-map.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/09-information-security/information-security-projects/secure-real-time-multiplayer-game.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/09-information-security/information-security-projects/secure-real-time-multiplayer-game.md)
`https://socket.io/`
- [x] [curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/map-the-debris.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/map-the-debris.md)
`http://en.wikipedia.org/wiki/Orbital_period`
- [x] [curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/dna-pairing.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/dna-pairing.md)
`http://en.wikipedia.org/wiki/Base_pair`
- [x] [curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/manipulating-complex-objects.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/manipulating-complex-objects.md)
`http://www.json.org/`
- [x] [curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/word-blanks.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/word-blanks.md)
`https://en.wikipedia.org/wiki/Mad_Libs`
- [x] [curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/counting-cards.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/counting-cards.md)
`https://en.wikipedia.org/wiki/Card_counting`
- [x] [curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/create-decimal-numbers-with-javascript.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/create-decimal-numbers-with-javascript.md)
`https://en.wikipedia.org/wiki/Floating_point#Accuracy_problems`
- [x] [curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/constructing-strings-with-variables.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/constructing-strings-with-variables.md)
`https://en.wikipedia.org/wiki/Mad_Libs`
- [x] [curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/golf-code.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/golf-code.md)
`https://en.wikipedia.org/wiki/Golf`
- [x] [curriculum/challenges/english/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/caesars-cipher.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/caesars-cipher.md)
`https://en.wikipedia.org/wiki/ROT13`
- [x] [curriculum/challenges/english/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/roman-numeral-converter.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/roman-numeral-converter.md)
`http://www.mathsisfun.com/roman-numerals.html`
- [x] [curriculum/challenges/english/03-front-end-libraries/front-end-libraries-projects/build-a-random-quote-machine.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/03-front-end-libraries/front-end-libraries-projects/build-a-random-quote-machine.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/03-front-end-libraries/front-end-libraries-projects/build-a-25-5-clock.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/03-front-end-libraries/front-end-libraries-projects/build-a-25-5-clock.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/03-front-end-libraries/front-end-libraries-projects/build-a-drum-machine.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/03-front-end-libraries/front-end-libraries-projects/build-a-drum-machine.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/03-front-end-libraries/front-end-libraries-projects/build-a-markdown-previewer.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/03-front-end-libraries/front-end-libraries-projects/build-a-markdown-previewer.md)
`https://en.wikipedia.org/wiki/User_story`
- [x] [curriculum/challenges/english/03-front-end-libraries/front-end-libraries-projects/build-a-javascript-calculator.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/03-front-end-libraries/front-end-libraries-projects/build-a-javascript-calculator.md)
`https://en.wikipedia.org/wiki/User_story`
- [ ] [curriculum/challenges/english/03-front-end-libraries/bootstrap/add-font-awesome-icons-to-our-buttons.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/03-front-end-libraries/bootstrap/add-font-awesome-icons-to-our-buttons.md)
`https://use.fontawesome.com/releases/v5.8.1/css/all.css`
- [x] [curriculum/challenges/english/03-front-end-libraries/react-and-redux/moving-forward-from-here.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/03-front-end-libraries/react-and-redux/moving-forward-from-here.md)
`https://github.com/facebookincubator/create-react-app`
- [x] [curriculum/challenges/english/03-front-end-libraries/react/set-state-with-this.setstate.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/03-front-end-libraries/react/set-state-with-this.setstate.md)
`https://facebook.github.io/react/docs/state-and-lifecycle.html`
- [x] [curriculum/challenges/english/03-front-end-libraries/react/use-the-lifecycle-method-componentwillmount.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/03-front-end-libraries/react/use-the-lifecycle-method-componentwillmount.md)
`https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html`
- [x] [curriculum/challenges/english/03-front-end-libraries/react/use-proptypes-to-define-the-props-you-expect.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/03-front-end-libraries/react/use-proptypes-to-define-the-props-you-expect.md)
`https://reactjs.org/docs/jsx-in-depth.html#specifying-the-react-element-type`
- [x] [curriculum/challenges/english/08-data-analysis-with-python/data-analysis-with-python-course/introduction-to-data-analysis.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/08-data-analysis-with-python/data-analysis-with-python-course/introduction-to-data-analysis.md)
`https://docs.google.com/presentation/d/1fDpjlyMiOMJyuc7_jMekcYLPP2XlSl1eWw9F7yE7byk`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sedols.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sedols.md)
`https://en.wikipedia.org/wiki/SEDOL`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/averages-pythagorean-means.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/averages-pythagorean-means.md)
`https://en.wikipedia.org/wiki/Pythagorean%20means`
`https://rosettacode.org/wiki/Averages/Arithmetic`
`https://en.wikipedia.org/wiki/Geometric`
`https://en.wikipedia.org/wiki/Harmonic`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/discordian-date.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/discordian-date.md)
`https://en.wikipedia.org/wiki/Gregorian%20calendar`
`https://en.wikipedia.org/wiki/Discordian%20calendar`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/cramers-rule.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/cramers-rule.md)
`https://en.wikipedia.org/wiki/linear%20algebra`
`https://en.wikipedia.org/wiki/Cramer&#x27;s%20rule`
`https://en.wikipedia.org/wiki/system%20of%20linear%20equations`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/cumulative-standard-deviation.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/cumulative-standard-deviation.md)
`https://en.wikipedia.org/wiki/Standard%20Deviation`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/hash-from-two-arrays.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/hash-from-two-arrays.md)
`https://rosettacode.org/wiki/Associative`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/linear-congruential-generator.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/linear-congruential-generator.md)
`https://en.wikipedia.org/wiki/linear%20congruential%20generator`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/self-describing-numbers.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/self-describing-numbers.md)
`https://en.wikipedia.org/wiki/Self-descriptive_number`
- [x] ~~[curriculum/challenges/english/10-coding-interview-prep/rosetta-code/euler-method.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/euler-method.md)
`https://en.wikipedia.org/wiki/Euler%20method`~~
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/i-before-e-except-after-c.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/i-before-e-except-after-c.md)
`https://en.wikipedia.org/wiki/I%20before%20E%20except%20after%20C`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/compare-a-list-of-strings.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/compare-a-list-of-strings.md)
  https://en.wikipedia.org/wiki/List_(abstract_data_type  
- [x] ~~[curriculum/challenges/english/10-coding-interview-prep/rosetta-code/babbage-problem.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/babbage-problem.md)
`https://en.wikipedia.org/wiki/Charles_Babbage`~~
- [x] ~~[curriculum/challenges/english/10-coding-interview-prep/rosetta-code/longest-common-subsequence.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/longest-common-subsequence.md)
`http://en.wikipedia.org/wiki/Longest_common_subsequence_problem`~~
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/s-expressions.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/s-expressions.md)
`https://en.wikipedia.org/wiki/S-Expression`
- [x] ~~[curriculum/challenges/english/10-coding-interview-prep/rosetta-code/closest-pair-problem.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/closest-pair-problem.md)
`https://en.wikipedia.org/wiki/Closest%20pair%20of%20points%20problem`
`https://en.wikipedia.org/wiki/Closest%20pair%20of%20points%20problem#Planar_case`
`https://en.wikipedia.org/wiki/Closest`
`https://www.cs.mcgill.ca/~cs251/ClosestPair/ClosestPairDQ.html`
`https://www.cs.ucsb.edu/~suri/cs235/ClosestPair.pdf`
`https://classes.cec.wustl.edu/~cse241/handouts/closestpair.pdf`~~
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/levenshtein-distance.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/levenshtein-distance.md)
`https://en.wikipedia.org/wiki/string%20metric`
`https://en.wikipedia.org/wiki/edit%20distance`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/fractran.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/fractran.md)
`https://en.wikipedia.org/wiki/FRACTRAN`
`https://en.wikipedia.org/wiki/John%20Horton%20Conway`
- [x] ~~[curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sailors-coconuts-and-a-monkey-problem.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sailors-coconuts-and-a-monkey-problem.md)
  https://www.youtube.com/watch?v=U9qU20VmvaU 
`https://oeis.org/A002021`~~
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/execute-a-markov-algorithm.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/execute-a-markov-algorithm.md)
`https://en.wikipedia.org/wiki/Markov%20algorithm`
`http://en.wikipedia.org/wiki/Turing_machine`
`http://en.wikipedia.org/wiki/Busy_beaver`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/farey-sequence.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/farey-sequence.md)
`https://en.wikipedia.org/wiki/Farey%20sequence`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/look-and-say-sequence.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/look-and-say-sequence.md)
`https://en.wikipedia.org/wiki/Look%20and%20say%20sequence`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/amicable-pairs.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/amicable-pairs.md)
`https://en.wikipedia.org/wiki/Amicable%20numbers`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-bead-sort.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-bead-sort.md)
  https://en.wikipedia.org/wiki/Bead_sort 
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sutherland-hodgman-polygon-clipping.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sutherland-hodgman-polygon-clipping.md)
  https://en.wikipedia.org/wiki/Sutherland-Hodgman%20clipping%20algorithm`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-comb-sort.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-comb-sort.md)
`https://en.wikipedia.org/wiki/Comb`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-shell-sort.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-shell-sort.md)
  https://en.wikipedia.org/wiki/Shell%20sort`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/lu-decomposition.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/lu-decomposition.md)
`https://en.wikipedia.org/wiki/LU%20decomposition`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/symmetric-difference.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/symmetric-difference.md)
  https://en.wikipedia.org/wiki/Symmetric%20difference`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/knights-tour.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/knights-tour.md)
`https://en.wikipedia.org/wiki/Knight%27s_tour`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/cut-a-rectangle.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/cut-a-rectangle.md)
`https://www.w3.org/2000/svg`
`https://www.w3.org/1999/xlink`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-stooge-sort.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-stooge-sort.md)
`https://en.wikipedia.org/wiki/Stooge%20sort`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/gray-code.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/gray-code.md)
`https://en.wikipedia.org/wiki/Gray%20code`
`https://en.wikipedia.org/wiki/Karnaugh%20map`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/hofstadter-q-sequence.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/hofstadter-q-sequence.md)
`https://en.wikipedia.org/wiki/Hofstadter_sequence#Hofstadter_Q_sequence`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/averages-mode.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/averages-mode.md)
`https://en.wikipedia.org/wiki/Mode%20(statistics)`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/happy-numbers.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/happy-numbers.md)
`https://en.wikipedia.org/wiki/Happy_number`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/lzw-compression.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/lzw-compression.md)
`https://en.wikipedia.org/wiki/Lempel-Ziv-Welch`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/straddling-checkerboard.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/straddling-checkerboard.md)
`https://en.wikipedia.org/wiki/Straddling_checkerboard`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/dot-product.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/dot-product.md)
`https://en.wikipedia.org/wiki/Dot%20product`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-strand-sort.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-strand-sort.md)
`https://en.wikipedia.org/wiki/Strand%20sort`
- [x] ~~[curriculum/challenges/english/10-coding-interview-prep/rosetta-code/hailstone-sequence.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/hailstone-sequence.md)
`https://en.wikipedia.org/wiki/Collatz%20conjecture`
`https://xkcd.com/710'~~
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-pancake-sort.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-pancake-sort.md)
`https://en.wikipedia.org/wiki/Pancake%20sorting`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/long-multiplication.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/long-multiplication.md)
`https://en.wikipedia.org/wiki/long%20multiplication`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/jortsort.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/jortsort.md)
`https://www.youtube.com/watch?v=pj4U_W0OFoE`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/kaprekar-numbers.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/kaprekar-numbers.md)
`https://en.wikipedia.org/wiki/Kaprekar%20number`
`https://oeis.org/A006886`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/soundex.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/soundex.md)
`https://en.wikipedia.org/wiki/soundex`
`https://www.archives.gov/research/census/soundex.html`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/deal-cards-for-freecell.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/deal-cards-for-freecell.md)
`https://freecellgamesolutions.com/`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/taxicab-numbers.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/taxicab-numbers.md)
`https://en.wikipedia.org/wiki/Hardy%E2%80%93Ramanujan%20number`
`https://oeis.org/A001235`
`https://en.wikipedia.org/wiki/Taxicab_number`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-cocktail-sort.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-cocktail-sort.md)
`https://en.wikipedia.org/wiki/Cocktail%20sort`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/longest-increasing-subsequence.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/longest-increasing-subsequence.md)
`https://en.wikipedia.org/wiki/Longest%20increasing%20subsequence`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/state-name-puzzle.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/state-name-puzzle.md)
`https://www.npr.org/templates/story/story.php?storyId=9264290`
`https://tapestry.tucson.az.us/twiki/bin/view/Main/StateNamesPuzzle`
`https://en.wikipedia.org/wiki/Goedel_numbering`
`https://en.wikipedia.org/wiki/Equivalence_relation`
`https://en.wikipedia.org/wiki/Equivalence_classes`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/y-combinator.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/y-combinator.md)
`https://en.wikipedia.org/wiki/Functional%20programming`
`https://en.wikipedia.org/wiki/lambda%20calculus`
`https://mvanier.livejournal.com/2897.html`
`https://en.wikipedia.org/wiki/Fixed-point%20combinator`
`https://en.wikipedia.org/wiki/Factorial`
`https://vimeo.com/45140590`
- [x] ~~[curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sha-256.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sha-256.md)
`https://en.wikipedia.org/wiki/SHA-256`~~
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/luhn-test-of-credit-card-numbers.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/luhn-test-of-credit-card-numbers.md)
`https://en.wikipedia.org/wiki/Luhn%20algorithm`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/heronian-triangles.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/heronian-triangles.md)
`https://en.wikipedia.org/wiki/Heron&#x27;s%20formula`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/fibonacci-word.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/fibonacci-word.md)
`https://hal.archives-ouvertes.fr/docs/00/36/79/72/PDF/The_Fibonacci_word_fractal.pdf`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-bogosort.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sorting-algorithms-bogosort.md)
`https://en.wikipedia.org/wiki/Bogosort`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/towers-of-hanoi.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/towers-of-hanoi.md)
`https://en.wikipedia.org/wiki/Towers_of_Hanoi`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sum-of-a-series.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sum-of-a-series.md)
`https://en.wikipedia.org/wiki/Series%20(mathematics)`
`https://en.wikipedia.org/wiki/sequence`
`https://en.wikipedia.org/wiki/Riemann%20zeta%20function`
`https://en.wikipedia.org/wiki/Basel%20problem`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/subleq.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/subleq.md)
`https://en.wikipedia.org/wiki/One_instruction_set_computer`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/accumulator-factory.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/accumulator-factory.md)
`https://en.wikipedia.org/wiki/Paul_Graham_(programmer)`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/fibonacci-n-step-number-sequences.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/fibonacci-n-step-number-sequences.md)
`https://en.wikipedia.org/wiki/Number%20prefix#Greek_series`
`https://en.wikipedia.org/wiki/Lucas%20number`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/count-the-coins.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/count-the-coins.md)
`https://en.wikipedia.org/wiki/United_States`
- [x] #46475
`https://oeis.org/A005228`
`https://oeis.org/A030124`
`https://en.wikipedia.org/wiki/Hofstadter_sequence#Hofstadter_Figure-Figure_sequences`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sudoku.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sudoku.md#L1)
`https://en.wikipedia.org/wiki/Sudoku`
`https://en.wikipedia.org/wiki/Algorithmics_of_sudoku`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/hash-join.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/hash-join.md)
`https://en.wikipedia.org/wiki/Join_(SQL)#Inner_join`
`https://en.wikipedia.org/wiki/Nested%20loop%20join`
`https://en.wikipedia.org/wiki/hash%20join`
`https://en.wikipedia.org/wiki/Multimap`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/comma-quibbling.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/comma-quibbling.md)
`https://blogs.msdn.com/b/ericlippert/archive/2009/04/15/comma-quibbling.aspx`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/averages-root-mean-square.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/averages-root-mean-square.md)
`https://en.wikipedia.org/wiki/Root%20mean%20square`
- [x] ~~[curriculum/challenges/english/10-coding-interview-prep/rosetta-code/zig-zag-matrix.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/zig-zag-matrix.md)
`https://en.wiktionary.org/wiki/antidiagonal`~~
- [x] ~~[curriculum/challenges/english/10-coding-interview-prep/rosetta-code/josephus-problem.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/josephus-problem.md)
`https://en.wikipedia.org/wiki/Josephus%20problem`~~
- [x] ~~[curriculum/challenges/english/10-coding-interview-prep/rosetta-code/topological-sort.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/topological-sort.md)
`https://en.wikipedia.org/wiki/Topological%20sorting`
`https://en.wikipedia.org/wiki/VHDL`
`https://en.wikipedia.org/wiki/Topological`
`https://www.embeddedrelated.com/showarticle/799.php`~~
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sort-stability.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sort-stability.md)
`https://en.wikipedia.org/wiki/Stable_sort#Stability`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/24-game.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/24-game.md)
`https://en.wikipedia.org/wiki/24_Game`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/iban.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/iban.md)
`https://en.wikipedia.org/wiki/International_Bank_Account_Number`
`https://en.wikipedia.org/wiki/Transcription_error`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/gamma-function.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/gamma-function.md)
`https://en.wikipedia.org/wiki/Gamma%20function`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/9-billion-names-of-god-the-integer.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/9-billion-names-of-god-the-integer.md)
`https://en.wikipedia.org/wiki/The%20Nine%20Billion%20Names%20of%20God#Plot_summary`
- [x] [curriculum/challenges/english/10-coding-interview-prep/rosetta-code/ludic-numbers.md](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/ludic-numbers.md)
`https://oeis.org/wiki/Ludic_numbers`

</details>

**Comment by nhcarrigan** [comment]
Personally, I'm a fan of internalising this information as much as possible - either in the challenge description, or as a news article for longer content.

This'll also help ensure that anything that changes (such as documentation updates that include breaking changes) won't affect our content.

**Comment by leungkimho** [comment]
I've been doing a random Rosetta Code challenge almost every day for the last few weeks and can provide some insight.

For the Rosetta Code challenges with external links, they've been hit and miss.  I had one just yesterday ([sum of a series](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/sum-of-a-series.md)) where the external links provided additional context to the problem, while I've had others where the original Rosetta Code description just wasn't enough without the external link ([execute a Markov algorithm](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/rosetta-code/execute-a-markov-algorithm.md)).

My observation from doing the challenges is the Rosetta Code content was pulled directly from Rosetta Code and parsed as is to work on the freeCodeCamp platform.  They have a different feel to them than other challenges written exclusively for freeCodeCamp.  Some of the descriptions fit perfectly with the spirit of the platform while other descriptions are so sparse the external link is needed.  I think it depends on each individual challenge what option to take.  I would lean towards rewriting the challenge so the link is not needed and users can stay on the platform, but there will be some that a rewrite would make the description too long to read.

Of course all of this is contingent on how much time and help your wonderful volunteers have and what type of process is wanted to mange each review of the descriptions and external links.

**Comment by ieahleen** [comment]
I think it depends

if it's about programming concepts it makes sense to incorporate them in the challenge or link to news articles

If it's about context, for example the golf challenge, having a news article about golf is a bit out of scope, there a link to an encyclopedic article from a site in the target language (like https://zh.wikipedia.org/wiki/%E9%AB%98%E7%88%BE%E5%A4%AB%E7%90%83 , I hope it's the right page) should be enough

the orbital period challenge in the JavaScript curriculum may need to have the meeded formula incorporated in the challenge description

**Comment by moT01** [comment]
They will probably need to be handled on a case by case basis. But I think the priority should be to:

1. If the info isn't necessary to complete the challenge, then I would just remove the link or hyperlink - this might include a slight rewording or removal of a sentence or two.
1. If the info seems necessary to complete the challenge - Try to bring the necessary info into the challenge and remove the link
1. If we can't remove the link with one of the above two methods, and there's enough info to make a news article out of it - we make an article and link to that. The articles can be translated and the links can go to their appropriate language.
1. If all else fails, keep the link - If we keep certain links, I would be fine with finding a substitute in the corresponding language. But I am thinking that we will likely be able to remove most of them with one of the first two options.

I am doubting that a news article would be necessary for any of the info, but I'm not sure.

It looks like the many on the list are from the rosetta code section - perhaps these should be kept to give credit. I would still prefer to remove them if we can, though.

**Comment by RandellDawson** [comment]
> perhaps these should be kept to give credit

@moT01 I only included Rosetta Code challenges that had links which were not links back to the Rosetta Code site.  The ones listed mainly have links to Wikipedia.

**Comment by moT01** [comment]
I didn't notice that - I vote to handle those like the rest then.

So we will just keep the links then that go to the rosetta site? I'm fine with that.

**Comment by MikeV2000** [comment]
I definitely agree with @moT01 suggestion. Considering that the external resource changes or is removed completely you would have to edit the challenge again anyway. So keeping the links as a reference could just be a way of stalling the problem.

**Comment by Shubham-Samanta** [comment]
I agree with @nhcarrigan all information should be internalized. It will be a fuzz at the beginning but  will be very helpful in the long run as it will not disrupt the user experience

**Comment by moT01** [comment]
I added a help wanted label. If anyone decides to work on this, I would try and keep the PR's to a reasonable size. Maybe a section or two at a time.

**Comment by sidemt** [comment]
I noticed the link for "the Unicon Discussion Web" in [State name puzzle](https://www.freecodecamp.org/learn/coding-interview-prep/rosetta-code/state-name-puzzle) challenge is not working, and found this issue.
The URL not working is `https://tapestry.tucson.az.us/twiki/bin/view/Main/StateNamesPuzzle`

I was not sure what to do with it since the challenge seemed to be relying on that page for explanation.

**Comment by ojeytonwilliams** [comment]
Hey @sidemt it seems like that info is lost.  Unless someone can dig it up from somewhere, the best we can do is remove any references to it from our text.

**Comment by Sboonny** [comment]
**You should ignore this**
for documention:

`1`, and `2` issues aren't found.
I will send pr about `3` issue shortly.

**Comment by Sboonny** [comment]
I have closed this issue, because what's left is GitHub development link, which could be an issue of its own without this much noise.
